### DISCUSSION TITLE
Drive Items Analytics

### DISCUSSION THREAD

**Discussion by malvarezColabra** [microsoftgraph/msgraph-sdk-dotnet#2690] [General]
Hello to everyone. I´m working in a script in C# to scan all files in a given site.
I have test every call in the graph explorer and they work just fine, except one. In the graph explorer I have tested this call
`https://graph.microsoft.com/v1.0/drives/{drive-id}/items/{driveItem-id}/analytics/allTime?$expand=activities($select=activityDateTime)&$select=activities,access`

The result of that api in graph explorer is
`{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#microsoft.graph.itemActivityStat",
    "access": {
        "actionCount": 2,
        "actorCount": 2,
        "timeSpentInSeconds": 0
    },
    "activities": [
        {
            "activityDateTime": "2024-09-30T18:56:56Z"
        },
        {
            "activityDateTime": "2024-04-16T21:39:27Z"
        },
        {
            "activityDateTime": "2024-04-16T21:38:37Z"
        }
    ]
}`

When I test this in C# the activity returns null
![image](https://github.com/user-attachments/assets/7acf5e42-27f7-42c1-ac9e-e3e1705d25dc)

Anyone have an idea why this is not returning data?

**Comment by andrueastman** [comment]
Thanks for raising this @malvarezColabra 

Any chance you can also share the code sample you are using to make the request with the SDK?

**Reply by malvarezColabra** [reply]
My apologies i didn't get the notification email of your answer
This is the function that I use to make de call, as you can see, have nothing special (I added the select to the activity, but this also works fine in the explorer).

The feature is enabled in the Site, the only thing that I can comment is that the folder is the shared documents in a team site. If you can confirm that I can't access to the information in that library, I will be considered as a valid answer.
```
private static async Task<ItemActivityStat> GetFileAnalytics(string itemId, string driveId)
{
    var result = await graphServiceClient.Drives[driveId].Items[itemId].Analytics.AllTime.GetAsync((requestConfiguration) =>
    {
        requestConfiguration.QueryParameters.Expand = ["activities($select=activityDateTime,actor)"];
        requestConfiguration.QueryParameters.Select = ["activities", "access"];
    });
    if (result == null)
        return null;

    return result;
}
```

**Reply by andrueastman** [reply]
Out of curiosity,
- Are you making the request to the same item on Graph Explorer and the C# code?
- How are you setting up the authentication for the C# application? Are you logging in as yourself with the same account as graph explorer?

**Reply by malvarezColabra** [reply]
Hello, yes the same drive, the same item.
The authentication in graph explorer is with my account but the application is with Azure Application with Sites.FullControl.All with the next code
```
try
{
    var scopes = new[] { "https://graph.microsoft.com/.default" };
    var options = new ClientSecretCredentialOptions
    {
        AuthorityHost = AzureAuthorityHosts.AzurePublicCloud
    };
    var clientSecretCredential = new ClientSecretCredential(tenantId, clientId, clientSecret, options);
    graphServiceClient = new GraphServiceClient(clientSecretCredential, scopes);
}
catch (Exception ex)
{
    throw new Exception("GetTokenAsync -> " + ex.Message);
}
```
